pkg/analytics/analytics.go

```diff
@@ -1,68 +1,70 @@
 package analytics
 
 import (
-	"os"
 	"time"
 
-	"github.com/AlecAivazis/survey/v2"
 	"github.com/airplanedev/cli/pkg/cli"
 	"github.com/airplanedev/cli/pkg/conf"
 	"github.com/airplanedev/cli/pkg/logger"
+	"github.com/airplanedev/cli/pkg/utils"
+	"github.com/airplanedev/cli/pkg/version"
 	"github.com/getsentry/sentry-go"
 	"gopkg.in/segmentio/analytics-go.v3"
 )
 
+// Set by Go Releaser.
 var (
 	segmentClient   analytics.Client
 	segmentWriteKey string
 	sentryDSN       string
 )
 
-func Init() error {
+func Init(cfg *cli.Config) error {
 	c, err := conf.ReadDefault()
 	if err != nil {
 		return err
 	}
-	if c.EnableTelemetry == "" {
+	if c.EnableTelemetry == nil {
 		// User has not specified one way or the other, ask them to opt-in.
 		if err := telemetryOptIn(c); err != nil {
 			return err
 		}
 		// Now try again.
-		return Init()
+		return Init(cfg)
 	}
-	if c.EnableTelemetry != "yes" {
+	if !*c.EnableTelemetry && !cfg.WithTelemetry {
 		return nil
 	}
-	segmentClient = analytics.New(segmentWriteKey)
+	if cfg.WithTelemetry && !*c.EnableTelemetry {
+		logger.Warning("Enabling usage analytics and error reports because --with-telemetry was set.")
```

Comment:
Maybe emphasize this is temporary, not permanent:

```suggestion
		logger.Warning("Temporarily enabling usage analytics and error reports because --with-telemetry was set.")
```